.github/workflows/build-and-push-docker-image.yml

```diff
@@ -0,0 +1,53 @@
+---
+
+name: 🏗️
+
+on:   # yamllint disable-line rule:truthy
+  pull_request:
+  push:
+    branches: ["release/*", "unstable/*"]
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Docker image tag (optional, defaults to Git ref)
+        required: false
```

Comment:
I think this should be required. I don't want the version to be guessed as there's no good heuristics until we have change notes that would indicate breaking changes or something.

Reply:
The version isn't "guessed," it's in the name of the Git tag that's used to trigger the workflow.

But sure, if you want to type the version number again, we can make that field required.

Also regarding https://github.com/pypa/gh-action-pypi-publish/pull/230#discussion_r1759783982, I'm still not sure what you're looking for here.

Reply:
@br3ndonland with `workflow_dispatch`, the version isn't available since it's not triggered by tag manipulations and so it must be an input. I'm specifically avoiding tags being the triggers. So the process is inverted. But it looks like you're already doing that (I haven't yet checked the new commits).